flink-contrib/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -911,9 +915,11 @@ void releaseResources(boolean canceled) {
 			if (canceled) {
 				List<StateObject> statesToDiscard = new ArrayList<>();
```

Comment:
We could also preallocate the size of `statesToDiscard` as the sum all things we are planning to insert in the next step.